File: Code/1672/1672.cpp
/******************************************************************************
 * 文件名: 1672
 * 作者: BOLUN XU
 * 创建日期: 2025
 * 版本: 1.0
 * 描述: 1672。
 * 时间复杂度： O(N)
 * 空间复杂度： O(N)
 * 执行用时： 3 ms
 * 消耗内存： 11.3 mb
 ******************************************************************************/
#include <iostream>
#include <algorithm>
#include <vector>
#include <numeric>

using namespace std;

class Solution {
    public:
    int maximumWealth(vector<vector<int>>& accounts) {
        std::vector<int> account_sum;
        for (const auto& account : accounts) {
            int sum = std::accumulate(account.begin(), account.end(), 0);
            account_sum.push_back(sum);
        }
        return *std::max_element(account_sum.begin(), account_sum.end());
    }
};

int main() {
    Solution sol;
    std::vector<std::vector<int>> accounts = {{1, 2, 3}, {3, 2, 1}};
    std::cout << sol.maximumWealth(accounts) << std::endl;
    return 0;
}

File: Code/1672/1672.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
文件名: PersonalLeetCode
作者: Bolun Xu
创建日期: 2025
版本: 1.0
描述: 1672。
时间复杂度： O(N)
空间复杂度： O(N)
执行用时： 0 ms
消耗内存： 17.6 mb
"""
from typing import List


class Solution:
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        account_sum = []
        for account in accounts:
            account_sum.append(sum(account))
        return max(account_sum)


if __name__ == '__main__':
    sol = Solution()
    print(sol.maximumWealth(accounts=[[1, 2, 3], [3, 2, 1]]))
